chore: use gatsby-parallel-runner

jlengstorf opened the pull request:
This will tell Netlify to build the site using `gatsby-parallel-runner` for faster builds!

If this deploy preview builds, we know it’s working 😂

jlengstorf commented:
hmm... something went weird with the Cypress download here, which I don't think is related to this PR?

kentcdodds commented on package.json:
```diff
@@ -123,7 +124,7 @@
     "start": "gatsby serve --port 5020",
     "setup": "npm install && npm run validate",
-    "netlify": "npm-run-all --parallel functions:build lint test netlify:cypress",
+    "netlify": "npm-run-all --parallel gatsby-parallel-runner lint test netlify:cypress",
```
I need my netlify functions to build. I don't think this is what I want...

kentcdodds replied:
Are you trying to replace my gatsby build with gatsby-parallel-runner?

jlengstorf replied:
Netlify will build the functions automatically as long as it knows which folder to look in — let me double-check that your `netlify.toml` is pointed there

jlengstorf replied:
confirmed in a local run that your functions will build without the `netlify-lambda` dep 🎉 

<img width="146" alt="Screen Shot 2020-02-26 at 4 46 58 PM" src="https://user-images.githubusercontent.com/163561/75401678-bc526f80-58b7-11ea-9ea6-fda25d37636e.png">

jlengstorf replied:
oh, forgot to update: the functions are deploying without `netlify-lambda` now; there was a minor path change required in `netlify.toml` to get that working

jlengstorf commented:
I got this working locally, but I'm waiting on a quick patch to get shipped, so I'm going to mark this WIP until then

jlengstorf commented:
@kentcdodds this is working now!

under the hood, we did three things:
1. we’re now persisting the Gatsby `.cache` and `public` folders between builds, which will have a huge benefit on repeat builds
2. switched the build command from `gatsby build` to `gatsby-parallel-runner`
3. added two env vars to your Netlify settings in support of this

kentcdodds commented:
Awesome! Thanks so much for this Jason!
